src/main/__tests__/authorize.spec.ts

```diff
@@ -0,0 +1,292 @@
+import fetchMock from 'jest-fetch-mock'
+
+import {
+  defineWindowProperty,
+  expectIframeWithParamss,
+  mockWindowCrypto
+} from './helpers/testHelpers'
+import { mockNextRandom, popNextRandomString } from './helpers/randomStringMock'
+import { randomBase64String } from '../../utils/random'
+import {
+  code,
+  confidential,
+  email,
+  getExpectedQueryString,
+  pageDisplay,
+  pblic,
+  phone,
+  tkn,
+  token, webMessage
+} from './helpers/oauthHelpers'
+import { createDefaultTestClient } from './helpers/clientFactory'
+import { LoginWithPasswordParams } from '../apiClient'
+import { loginWithPasswordTest } from './loginWithPassword.spec'
+import { signupTest } from './signup.spec'
+
+fetchMock.enableMocks()
+defineWindowProperty('crypto', mockWindowCrypto)
+defineWindowProperty('location')
+
+beforeEach(() => {
+  document.body.innerHTML = ''
+  jest.resetAllMocks()
+  fetchMock.resetMocks()
+  popNextRandomString()
+})
+
+describe('with redirection', () => {
+
+  describe.each`
+  clientType            | responseType
+  ${pblic}          | ${token}
```

Comment:
`pblic` is intentional as `public` is a reserved keyword... suggestions accepted 😄

Reply:
Oh, maybe `pυblic` ? 🇬🇷

Reply:
Not sure I want to use a special char like "υ" :/